platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -292,14 +304,43 @@ private void captureTimestamp(final ConsensusEvent event) {
      *
      * @param transaction the transaction to apply
      */
-    private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
-            return;
+    private ConsensusTransaction handleTransaction(final ConsensusTransaction transaction) {
+        if (isSystemTransaction(transaction)) {
+            return transaction;
         }
+
         final int delta =
                 ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
+
+        return null;
+    }
+
+    private boolean isSystemTransaction(final ConsensusTransaction transaction) {
```

Comment:
we should not pass the old system transactions to the callback since the platform will handle these, we should only invoke it for the new transactions

Reply:
Good catch! I updated the logic.